### ISSUE TITLE
MMEX + Syncthing

### ISSUE THREAD

**Issue by iandlg** [moneymanagerex/android-money-manager-ex#1705]
With syncthing I have a linked folder between my computer (windows) and my android phone. When I update the .mmb database file using the desktop app I get the change almost instantly on my phone. But when I try to change the database with the Android app the file isn't modified.

**To Reproduce**
1. Create .mmb on desktop
2. Open the same .mmb on android
3. Enable synchronization on android
4. Make a change to .mmb on android
5. Click on synchronize, upload and download 
6. Close android app
7. Wait for the change to appear on desktop

The change never happens, from the modified time I can also tell that the android app is not make any modifications.
 
 - Device: Redmi note 10 pro
 - OS: miui 13
 - Storage: Phone storage + Syncthing
 - Synchronization: Enabled with 5min interval and Sync on Start
 - Encrypted: no
 - App Version: 2024.05.02

**Comment by guanlisheng** [comment]
hi @iandlg, thanks for trying mmex 
if `Enabled with 5min interval and Sync on Start` is your synchronization setting, i would like you to try upgrading it to the latest version:  `2024.05.09`

**Comment by iandlg** [comment]
I tried updating but there is still the same problem.

**Comment by guanlisheng** [comment]
that's wired as desktop -> android is ok 
Can you share your detailed synchronization setting for both MMEX and Syncthing?

**Comment by iandlg** [comment]
Yeah so on here are the settings for the syncthing file
![image](https://github.com/moneymanagerex/android-money-manager-ex/assets/88088086/be2c64cf-96d4-4367-a292-3a05133d572b)
And here are the ones for mmex
![Screenshot_2024-06-02-19-12-25-154_com money manager ex](https://github.com/moneymanagerex/android-money-manager-ex/assets/88088086/02cadf77-b6d5-4ce3-9cc0-05560faab6b6)
And the update frequency is at 5min

**Comment by guanlisheng** [comment]
Thanks. everything seems ok. Syncthing should take the changes from Android mmex every 5m. 
can you add a trx in your android mmex, and sync it.  then check the database file in android folder?

**Comment by iandlg** [comment]
A transaction you mean?

**Comment by iandlg** [comment]
So i just added a transaction from android and it says "Your database has been stored successfully" and "the database is up to date". Then I closed the app. And when I go to the files on android the last modification was at 19:14 not 19:40.

**Comment by iandlg** [comment]
Yeah same file in both. Yup the transactions stays even when I close the app and open it up again. It just doesn't appear on the desktop.
Also when I manually reopen the .mmb from android, it deletes the transaction.

**Comment by iandlg** [comment]
![Screenshot_2024-06-02-20-01-47-689_com money manager ex](https://github.com/moneymanagerex/android-money-manager-ex/assets/88088086/67201866-178a-4f3c-8fe4-5f8e322c2292)
2. Ok so when I export the database from android and read it on desktop it works

**Comment by guanlisheng** [comment]
Thanks. 

`export` works, and it is a sort of manual push.

now, i have no idea why the synchronization was not working.

**Comment by guanlisheng** [comment]
hi @iandlg
what's your android syncthing version?

**Comment by rljt24** [comment]
Hi, I am having the same issue. I can push changes in my database from my desktop to my phone but not the other way around. I think the issue is that the database path on my phone is not going to my snyc'd folder but the path that you see in the picture above. I tried making a folder that used that path in Syncthing, but Android would not let me use that folder. Is there a way to change the database path?

**Comment by guanlisheng** [comment]
hi @rljt24,  please describe your detailed configuration and steps to re-produce the issue.

**Comment by raphaelsuzuki** [comment]
I faced the same issue when trying to sync MMEX on Android with Syncthing. It turns out that there's no support for SAF on Syncthing and probably won't be on the foreseeable future: https://github.com/syncthing/syncthing-android/issues/2047

So as long as MMEX is dependent on SAF to work on Android, users should look for other alternatives.

**Comment by guanlisheng** [comment]
thanks, @raphaelsuzuki for sharing.

**Comment by guanlisheng** [comment]
https://github.com/moneymanagerex/android-money-manager-ex/releases/tag/2025.01.03.1058 might help and going to close the issue
